sys/kern/kern_cheri_revoke.c

```diff
@@ -490,8 +543,14 @@ kern_cheri_revoke(struct thread *td, int flags,
 				PROC_UNLOCK(td->td_proc);
 
 				vm_map_lock(vmm);
+				/* Roll back some earlier state changes. */
 				cheri_revoke_st_set(&vmm->vm_cheri_revoke_st,
 				    epoch, entryst);
+				if (flags & CHERI_REVOKE_ASYNC) {
```

Comment:
Move the `cheri_reovke_st_set` above this `if` into an `else` clause?

Reply:
Sorry, but why? In the asynchronous case we still want to make sure that the main state is rolled back.

Reply:
No, sorry, I had wires crossed and was thinking this was the "publish to userspace" "set the revocation state".  Please disregard.